backend/Controllers/UserController.cs

```diff
@@ -83,6 +88,15 @@ public async Task<IActionResult> CreateUser([FromBody] NewUserResource newUser)
             return Created("Created new user successfully!", safeUser);
         }
 
+        [HttpGet("profile")]
+        public async Task<IActionResult> GetUser()
+        {
+            var users = await usersService.GetAllUsersAsync();
+            IEnumerable<SafeUserResource> safeUsers = mapper.Map<IEnumerable<User>, IEnumerable<SafeUserResource>>(users);
+            var targetUsername = httpContextAccessor.HttpContext.User.Identity.Name;
```

Comment:
What is this Httpcontext thing? Is it inbuilt in ASP?

Reply:
Yes

Reply:
It is added in startup line 87: "services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();"